javalin/src/main/java/io/javalin/http/Context.kt

```diff
@@ -424,10 +424,16 @@ interface Context {
     fun json(obj: Any): Context = json(obj, obj::class.java)
 
     /**
-     * Serializes object to a JSON-stream using the registered [io.javalin.json.JsonMapper] and sets it as the context result.
-     * Also sets content type to application/json.
+     * Serializes object to a JSON-stream using the registered [io.javalin.json.JsonMapper]. The registered
+     * [io.javalin.json.JsonMapper] may write the output stream directly, or it may return a value that will
+     * be set as the context result. Also sets content type to application/json.
      */
-    fun jsonStream(obj: Any, type: Type): Context = contentType(ContentType.APPLICATION_JSON).result(jsonMapper().toJsonStream(obj, type))
+    fun jsonStream(obj: Any, type: Type): Context = contentType(ContentType.APPLICATION_JSON).apply {
+        if (!jsonMapper().handleOutputStream(outputStream(), obj, type)) {
```

Comment:
Using `outputStream()` rather than `req().outputStream()`. This will use the compressed output stream, rather than the raw output stream.